gnomad_qc/v4/create_release/update_hgdp_tgp_subset_AF.py

```diff
@@ -0,0 +1,580 @@
+    related = hl.is_defined(relatedness_ht[ht.key])
+    # Update annotations impacted by the updated HGDP + 1KG metadata.
+    outlier = hl.is_defined(pop_outliers_ht[ht.key])
+    populations = populations_ht[ht.key]
+    # TODO: Will need to update based on gnomad_sample_filters.release_related decision.
+    gnomad_release = (
+        ~hard_filtered
+        & ~outlier
+        & ~related  # & ~ht.gnomad_sample_filters.release_related
+    )
+
+    sample_annotations_to_update = {
+        "bam_metrics": {"freemix": freemix},
+        "gnomad_sample_filters": {
+            "hard_filters": hard_filters,
+            "hard_filtered": hard_filtered,
+        },
+        "gnomad_high_quality": ht.gnomad_high_quality & ~hard_filtered,
+        "gnomad_release": gnomad_release,
+        "relatedness_inference": {"related": related},
+        "hgdp_tgp_meta": {
+            "subcontinental_pca": {"outlier": outlier},
+            "population": populations.population,
+            "latitude": populations.latitude,
+            "longitude": populations.longitude,
+        },
+        "high_quality": ~hard_filtered & ~outlier,
+    }
+    updated_ht = update_structured_annotations(
+        ht,
+        sample_annotations_to_update,
+        annotation_update_label="sample_annotations_updated",
+    )
+    updated_ht = updated_ht.checkpoint(
+        hl.utils.new_temp_file("hgdp_tgp_meta_update", extension="ht"), overwrite=True
+    )
+    updated_counts = updated_ht.aggregate(
+        hl.struct(
+            n_hard_filtered=hl.agg.count_where(
+                updated_ht.gnomad_sample_filters.hard_filtered
+            ),
+            n_related=hl.agg.count_where(updated_ht.relatedness_inference.related),
+            n_outlier=hl.agg.count_where(
+                updated_ht.hgdp_tgp_meta.subcontinental_pca.outlier
+            ),
+            n_diff=hl.agg.explode(
+                lambda x: hl.agg.counter(x), updated_ht.sample_annotations_updated
+            ),
+        )
+    )
+    logger.info(
+        """%d sample(s) hard filtered
+        %d more sample(s) hard filtered than last release
+        %d sample(s) related
+        %d sample(s) were assigned differently for the relatedness than last release
+        %d sample(s) are pop outliers
+        %d sample(s) were assigned differently as subcontinental outlier than last release
+        %d samples have different population labels than last release""",
+        updated_counts["n_hard_filtered"],
+        updated_counts["n_diff"]["gnomad_sample_filters.hard_filtered"],
+        updated_counts["n_related"],
+        updated_counts["n_diff"]["relatedness_inference.related"],
+        updated_counts["n_outlier"],
+        updated_counts["n_diff"]["hgdp_tgp_meta.subcontinental_pca.outlier"],
+        updated_counts["n_diff"]["hgdp_tgp_meta.population"],
+    )
+    return updated_ht
+
+
+def get_filtered_samples(ht: hl.Table) -> Tuple[hl.Table, hl.Table]:
```

Comment:
Adding for another suggestion below

```suggestion
def get_filtered_samples(ht: hl.Table) -> Tuple[hl.Table, hl.Table, hl.Table]:
```